Replace deprecated Py_Unicode_EncodeUTF8 in objToJSON.c

Warning when building the C extensions:

```
pandas/_libs/src/ujson/python/objToJSON.c:444:14: warning: 'PyUnicode_EncodeUTF8' is deprecated [-Wdeprecated-declarations]
    newObj = PyUnicode_EncodeUTF8(PyUnicode_AS_UNICODE(obj),
             ^
/usr/local/Cellar/python/3.7.1/Frameworks/Python.framework/Versions/3.7/include/python3.7m/unicodeobject.h:1324:7: note: 'PyUnicode_EncodeUTF8' has been
      explicitly marked deprecated here
    ) Py_DEPRECATED(3.3);
      ^
/usr/local/Cellar/python/3.7.1/Frameworks/Python.framework/Versions/3.7/include/python3.7m/pyport.h:493:54: note: expanded from macro 'Py_DEPRECATED'
#define Py_DEPRECATED(VERSION_UNUSED) __attribute__((__deprecated__))
 ```

https://github.com/pandas-dev/pandas/blob/91802fb0accde031c3b6aca040a8b533a193fef6/pandas/_libs/src/ujson/python/objToJSON.c#L436-L437

--- a/pandas/_libs/src/ujson/python/objToJSON.c
+++ b/pandas/_libs/src/ujson/python/objToJSON.c
@@ -433,8 +433,7 @@ static void *PyUnicodeToUTF8(JSOBJ _obj, JSONTypeContext *tc, void *outValue,
     }
 #endif
 
-    newObj = PyUnicode_EncodeUTF8(PyUnicode_AS_UNICODE(obj),
-                                  PyUnicode_GET_SIZE(obj), NULL);
+    newObj = PyUnicode_AsUTF8String(obj);
 
     GET_TC(tc)->newObj = newObj;
 

